### ISSUE TITLE
Invalid code in dtio_22.f90

### ISSUE THREAD

**Issue by nncarlson** [nncarlson/gfortran.dg#17]
The interface for the `WRITE(FORMATTED)` procedure does not conform to the required interface (see F08 9.6.4.8.3).  The NAG compiler catches this error. Here is a patch:
```patch
diff --git a/dtio_22.f90 b/dtio_22.f90
index d0fb76c..8e5a395 100644
--- a/dtio_22.f90
+++ b/dtio_22.f90
@@ -15,10 +15,10 @@ contains
   subroutine wf(this, unit, b, c, iostat, iomsg)
     class(t), intent(in) :: this
     integer, intent(in) :: unit
-    character, intent(in) :: b
+    character(*), intent(in) :: b
     integer, intent(in) :: c(:)
     integer, intent(out) :: iostat
-    character, intent(inout) :: iomsg
+    character(*), intent(inout) :: iomsg
     write (unit, "(i3)", IOSTAT=iostat, IOMSG=iomsg) this%i
   end subroutine
 end

**Comment by janusw** [comment]
See `check_dtio_interface1` and `check_dtio_arg_TKR_intent` in interface.c. Apparently we're checking TKR and INTENT, but miss out on the character length.

**Comment by janusw** [comment]
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=84409

**Comment by janusw** [comment]
Fixed with 2707b5aa89c4c75e03917d543474a482fec9808e.
